mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/TaskExecutorRegistration.java

```diff
@@ -113,20 +113,14 @@ public boolean containsAttributes(Map<String, String> requiredAttributes) {
     }
 
     @JsonIgnore
-    public Optional<ContainerSkuID> getTaskExecutorContainerDefinitionId() {
+    public ContainerSkuID getTaskExecutorContainerDefinitionId() {
         // handle back compat on key case insensitivity.
         String containerDefIdLower = WorkerConstants.WORKER_CONTAINER_DEFINITION_ID.toLowerCase();
         if (this.taskExecutorAttributes.containsKey(containerDefIdLower)) {
-            return Optional.ofNullable(ContainerSkuID.of(this.getTaskExecutorAttributes().get(containerDefIdLower)));
+            return ContainerSkuID.of(this.getTaskExecutorAttributes().get(containerDefIdLower));
```

Comment:
what's the release plan to prod master/agents for this breaking change?

Reply:
this is not changing anymore, so there is no problem during rollout.